osquery/tables/system/darwin/apps.mm

```diff
@@ -313,10 +313,22 @@ QueryData genAppSchemes(QueryContext& context) {
     }
 
     // Check the default handler assigned to the protocol scheme.
-#if !defined(DARWIN_10_9)
-    auto default_app =
-        LSCopyDefaultApplicationURLForURL(url, kLSRolesAll, nullptr);
-#endif
+    // This only applies to 10.10, so resolve the symbol at runtime.
+    CFBundleRef ls_bundle =
+        CFBundleGetBundleWithIdentifier(CFSTR("com.apple.LaunchServices"));
+    CFURLRef default_app = nullptr;
+    if (ls_bundle != nullptr) {
+      auto _LSCopyDefaultApplicationURLForURL =
+          (CFURLRef (*)(CFURLRef, LSRolesMask, CFErrorRef*))
+          CFBundleGetFunctionPointerForName(
+              ls_bundle, CFSTR("LSCopyDefaultApplicationURLForURL"));
+      // If the symbol did not exist we will not have a handle.
+      if (_LSCopyDefaultApplicationURLForURL != nullptr) {
+        default_app =
+            _LSCopyDefaultApplicationURLForURL(url, kLSRolesAll, nullptr);
+      }
+    }
+
```

Comment:
Slick indeed. Does this imply that we don't have to worry about #1446? IIRC apps and xattr are the only places where we had diverging code paths for 10.9 and 10.10.